spring-integration-core/src/test/java/org/springframework/integration/graph/IntegrationGraphServerTests.java

```diff
@@ -152,10 +150,17 @@ void test() throws Exception {
 		this.testSource.receive();
 		this.expressionRouterInput.send(MessageBuilder.withPayload("foo").setHeader("foo", "fizChannel").build());
 
+		jsonArray = JsonPathUtils.evaluate(baos.toByteArray(), "$..nodes[?(@.name == 'router')]");
+		String routerJson = jsonArray.toJSONString();
+		assertThat(routerJson).contains("\"deprecated\":\"stats are deprecated");
+
+
 		this.server.rebuild();
 		graph = this.server.getGraph();
 		baos = new ByteArrayOutputStream();
+		objectMapper = new ObjectMapper();
 		objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+		objectMapper.registerModule(new GraphLegacyStatsNullModule());
```

Comment:
How about this instead:

```
mapper.registerModule(new SimpleModule().addSerializer(IntegrationNode.Stats.class, NullSerializer.instance));
```
?

So, we don't need to have any custom classes in the framework.

Or `MixInAnnotations`...

Or let them decide what is the best Jackson option to exclude the property:

https://stackoverflow.com/questions/7421474/how-can-i-tell-jackson-to-ignore-a-property-for-which-i-dont-have-control-over